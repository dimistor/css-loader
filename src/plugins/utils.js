import postcss from 'postcss';

const createICSSImports = (imports) =>
  Object.keys(imports).map((path) => {
    const aliases = imports[path];
    const declarations = Object.keys(aliases).map((key) =>
      postcss.decl({
        prop: key,
        value: aliases[key],
        raws: { before: '\n  ' },
      })
    );

    return postcss
      .rule({
        selector: `:import(${path})`,
        raws: { after: '\n' },
      })
      .append(declarations);
  });

export default { createICSSImports };
