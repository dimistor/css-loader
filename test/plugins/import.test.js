import fs from 'fs';
import path from 'path';

import plugin from '../../src/plugins/import';
import { runPostcss } from '../helpers/utils';

describe('import postcss plugin', () => {
  it('basic', async () => {
    const result = await runPostcss(
      fs.readFileSync(path.join(__dirname, '../fixtures/import/import.css')),
      [plugin]
    );

    expect(result.css).toMatchSnapshot('css');
  });
});
