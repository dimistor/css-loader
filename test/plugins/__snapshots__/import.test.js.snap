// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import postcss plugin basic: css 1`] = `
":import('test.css'){
}
:import('test.css' screen and print){
}
:import('test-media.css' screen and print){
}
:import('test-other.css' (min-width: 100px)){
}
:import('http://example.com/style.css'){
}
:import('http://example.com/other-style.css' screen and print){
}
:import('//example.com/style.css'){
}
:import('~package/test.css'){
}
:import('query.css?foo=1&bar=1'){
}
:import('other-query.css?foo=1&bar=1#hash'){
}
@import url();
@import url('');
@import url(\\"\\");
@import '';
@import \\"\\";
@import \\"   \\";
@import url();
@import url('');
@import url(\\"\\");
@import ;
@import foo-bar;
@import-normalize;
@import url('http://') :root {}

.class {
  a: b c d;
}

.foo {
  @import 'path.css';
}
"
`;
